recipe/meta.yaml

```diff
@@ -217,13 +223,22 @@ outputs:
         - {{ pin_subpackage('mkl-include', exact=True) }}
         - blas * mkl
     test:
+      requires:
+        - {{ compiler('c') }}
+        - cmake
+        - ninja
+        - pkg-config
+      files:
+        - tests
       commands:
         - IF NOT EXIST %PREFIX%\Library\lib\mkl*dll.lib EXIT /B 1  # [win]
         # Verify that the mkl library can be located through pkg-config
         - test -f `pkg-config --variable=libdir --dont-define-prefix mkl-sdl`/libmkl_rt${SHLIB_EXT}  # [unix]
         - for /f "usebackq tokens=*" %%a in (`pkg-config --variable=libdir --dont-define-prefix mkl-sdl`) do if not exist "%%a/mkl_rt.lib" exit 1  # [win]
-      requires:
-        - pkg-config
+        - cmake -S tests/mkl-devel -B build -G Ninja -DCMAKE_VERBOSE_MAKEFILE=ON  # [linux or win]
+        - cmake --build build -v                                                  # [linux or win]
+        - ./build/myapp                                                           # [linux]
+        - build\myapp                                                             # [win]
```

Comment:
I'm guessing this will be a failing test at runtime, returning some nonzero value, if libiomp5md.lib isn't found or isn't working properly? (ie. do you need to test for the string output in app.c?)

Reply:
The compiled app doesn't use OpenMP, so it wouldn't fail I think. This (and the cmake stuff) tests that the cmake config file works. As for `libiomp5md.lib`, it's only require at build time. COMpiled stuff will link against the DLL, not the import library.

I guess we could potentially add a check in the cmake file to make sure that `libiomp5md.lib` was correctly found.

Reply:
So I forcefully removed `libiomp` on Linux and ran CMake and it correctly errored out.

```
-- Found MKL: intelmpi  
-- Found MKL: /opt/conda/envs/test  
-- Found MKL: /opt/conda/envs/test/lib/libmkl_intel_ilp64.so  
-- Found MKL: /opt/conda/envs/test/lib/libmkl_intel_thread.so  
-- Found MKL: /opt/conda/envs/test/lib/libmkl_core.so  
CMake Error at /opt/conda/envs/test/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find MKL (missing: OMP_LIBRARY OMP_LINK)
Call Stack (most recent call first):
  /opt/conda/envs/test/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /opt/conda/envs/test/lib/cmake/mkl/MKLConfig.cmake:743 (find_package_handle_standard_args)
  CMakeLists.txt:7 (find_package)


-- Configuring incomplete, errors occurred!

```
So this answers part of your question. I'll see if I can add some OpenMP code in the sample app to really confirm that it can be used and linked and that it works.

Reply:
Actually, if I add `-Wl,-flat_namespace` to the linkerflags, it works.